source/material_model/rheology/visco_plastic.cc

```diff
@@ -128,6 +132,19 @@ namespace aspect
           edot_ii = std::max(std::sqrt(std::fabs(second_invariant(deviator(in.strain_rate[i])))),
                              min_strain_rate);
 
+        // In case porosity lies above the melt transport threshold
+        // P_effective = P_bulk - P_f = (1-porosity) * P_s + porosity * P_f - P_f = (1-porosity) * (P_s - P_f)
+        // otherwise,
+        // P_effective = P_bulk, which equals P_solid (which is given by in.pressure[i])
+        double effective_pressure = in.pressure[i], porosity = 0;
+
+        if (this->include_melt_transport() && this->get_melt_handler().is_melt_cell(in.current_cell))
```

Comment:
I wonder for now if we want to actually make this new functionality dependent on whether melt transport is included? For example, could we start with a general case where an additional pressure (of any origin) is passed to the visco plastic rheology model? This would make testing easier. However, maybe it makes sense to begin with having melt accounted for as well?